recommend/policyTeamplates.go

```diff
@@ -0,0 +1,241 @@
+		home := os.Getenv("HOMEDRIVE") + os.Getenv("HOMEPATH")
+		if home == "" {
+			home = os.Getenv("USERPROFILE")
+		}
+		return home
+	}
+	return os.Getenv("HOME")
+
+}
+
+// IsTemplates function returns error if policy-template folder not found
+func isTemplates() error {
+	path := fmt.Sprintf("%s/.cache/karmor/", userHome())
+
+	if stat, err := os.Stat(path); err == nil && stat.IsDir() {
+		err := isLatest()
+		if err != nil {
+			return err
+		}
+	} else {
+		log.WithFields(log.Fields{
+			"Current Version": "nil",
+		}).Info("policy-templates not found. Trying to download")
+		ver, err := downloadAndUnzipRelease()
+		if err != nil {
+			return err
+		}
+		log.WithFields(log.Fields{
+			"Current Version": ver,
+		}).Info("policy-templates download completed")
+	}
+	return nil
+
+}
+
+func latestRelease() (*github.RepositoryRelease, error) {
+	latestRelease, _, err := github.NewClient(nil).Repositories.GetLatestRelease(context.Background(), org, repo)
+	return latestRelease, err
+
+}
+
+func isLatest() error {
+	latestRelease, err := latestRelease()
+	if err != nil {
+		return err
+	}
+	path := fmt.Sprintf("%s/.cache/karmor/", userHome())
+	latestFolderName := fmt.Sprintf("%s%s-%s", path, repo, strings.TrimPrefix(*latestRelease.TagName, "v"))
+	files, err := os.ReadDir(path)
+	if err != nil {
+		return err
+	}
+	for _, file := range files {
+		currentFolderName := fmt.Sprintf("%s%s", path, file.Name())
+		if file.IsDir() && currentFolderName != latestFolderName {
+			return errors.New("policy-template version is outdate. Please use `karmor recommend --update` to update the policy-template to the latest version ")
+
+		}
+
+	}
+
+	return nil
+}
+
+func removeData(file string) error {
+
+	err := os.Remove(file)
+	return err
+}
+
+func downloadAndUnzipRelease() (string, error) {
+
+	latestRelease, err := latestRelease()
+	if err != nil {
+		return "", err
+	}
+	path := fmt.Sprintf("%s/.cache/karmor/", userHome())
+	_ = os.RemoveAll(path)
+	err = os.MkdirAll(filepath.Dir(path), 0750)
+	if err != nil {
+		return "", err
+	}
+
+	downloadURL := fmt.Sprintf("%s%s.zip", url, *latestRelease.TagName)
+
+	resp, err := grab.Get(path, downloadURL)
+	if err != nil {
+		return "", err
+	}
+	err = unZip(resp.Filename, path)
+	if err != nil {
+		return "", err
+	}
+
+	err = removeData(resp.Filename)
+	if err != nil {
+		return "", err
+	}
+	_ = updatePolicyRules(strings.TrimSuffix(resp.Filename, ".zip"))
+
+	return *latestRelease.TagName, nil
+
+}
+
+func unZip(source, dest string) error {
+	read, err := zip.OpenReader(source)
+	if err != nil {
+		return err
+	}
+	defer read.Close()
+	for _, file := range read.File {
+		if file.Mode().IsDir() {
+			continue
+		}
+		open, err := file.Open()
+		if err != nil {
+			return err
+		}
+		name, err := sanitizeArchivePath(dest, file.Name)
+		if err != nil {
+			return err
+		}
+		_ = os.MkdirAll(path.Dir(name), 0750)
+		create, err := os.Create(filepath.Clean(name))
+		if err != nil {
+			return err
+		}
+
```

Comment:
use defer create.Close() here .. otherwise it leads to leak below in the case of error situation